src/test/scala/chiselTests/reflect/DataMirrorSpec.scala

```diff
@@ -265,4 +265,40 @@ class DataMirrorSpec extends ChiselFlatSpec {
       foo.ports1 should be(Seq(foo.in, foo.out, foo.other))
     })
   }
+
+  "modulePorts" should "return an Instance of a module's IOs" in {
+    import chiselTests.experimental.hierarchy.Examples.AddOne
+    class Foo extends Module {
+      val definition = Definition(new AddOne)
+      val instA = Instance(definition)
+      val portsA = DataMirror.modulePorts(instA)
+
+      val instB = (Module(new AddOne)).toInstance
+      val portsB = DataMirror.modulePorts(instB)
+    }
+
+    ChiselStage.emitCHIRRTL(new Module {
+      val foo = Module(new Foo)
+      (foo.portsA.map(_._2) should contain).allOf(foo.instA.in, foo.instA.out)
+      (foo.portsB.map(_._2) should contain).allOf(foo.instB.in, foo.instB.out)
+    })
+  }
+
+  "fullModulePorts" should "return an Instance of a module's IOs" in {
+    import chiselTests.experimental.hierarchy.Examples.AddOne
+    class Foo extends Module {
+      val definition = Definition(new AddOne)
+      val instA = Instance(definition)
+      val portsA = DataMirror.fullModulePorts(instA)
+
+      val instB = (Module(new AddOne)).toInstance
+      val portsB = DataMirror.fullModulePorts(instB)
+    }
+
+    ChiselStage.emitCHIRRTL(new Module {
+      val foo = Module(new Foo)
+      (foo.portsA.map(_._2) should contain).allOf(foo.instA.in, foo.instA.out)
+      (foo.portsB.map(_._2) should contain).allOf(foo.instB.in, foo.instB.out)
+    })
+  }
```

Comment:
Generally looks good, but please use a Proto module type that has ports that illustrates the difference between `.modulePorts` and `.fullModulePorts` as for `AddOne` they give the same answer 🙂

Reply:
oops yes, can you check the new test?